scripts/o3de/tests/ExportScripts/test_export_source_built_project.py

```diff
@@ -0,0 +1,1296 @@
+                                  False,
+                                  "release",
+                                  "profile",
+                                  [],[],[])
+        
+        mock_is_sdk_engine.assert_called_once_with(engine_path=mock_ctx.engine_path)
+        mock_has_mono_artifacts.assert_called_once_with(mock_ctx)
+        mock_kill_processes.assert_not_called()
+
+
+@pytest.mark.parametrize("is_engine_centric, use_sdk, should_build_tools_flag, has_monolithic, use_monolithic, expect_toolchain_build_called", [
+    pytest.param(False, True, True, True, True,  False),
+    pytest.param(False, True, True, True, False, False),
+    pytest.param(False, True, True, False, True, False),
+    pytest.param(False, True, True, False, False, False),
+    pytest.param(False, True, False, False, False, False),
+    pytest.param(False, True, False, True, False, False),
+    pytest.param(False, True, False, False, True, False),
+    pytest.param(False, True, False, True, True, False),
+
+    pytest.param(False, False, True, True, True,  True),
+    pytest.param(False, False, True, True, False, True),
+    pytest.param(False, False, True, False, True, True),
+    pytest.param(False, False, True, False, False, True),
+    pytest.param(False, False, False, False, False, False),
+    pytest.param(False, False, False, True, False, False),
+    pytest.param(False, False, False, False, True, False),
+    pytest.param(False, False, False, True, True, False),
+
+    pytest.param(True, True, True, True, True,  False),
+    pytest.param(True, True, True, True, False, False),
+    pytest.param(True, True, True, False, True, False),
+    pytest.param(True, True, True, False, False, False),
+    pytest.param(True, True, False, False, False, False),
+    pytest.param(True, True, False, True, False, False),
+    pytest.param(True, True, False, False, True, False),
+    pytest.param(True, True, False, True, True, False),
+
+    pytest.param(True, False, True, True, True,  True),
+    pytest.param(True, False, True, True, False, True),
+    pytest.param(True, False, True, False, True, True),
+    pytest.param(True, False, True, False, False, True),
+    pytest.param(True, False, False, False, False, False),
+    pytest.param(True, False, False, True, False, False),
+    pytest.param(True, False, False, False, True, False),
+    pytest.param(True, False, False, True, True, False),
+])
+def test_build_tools_combinations(tmp_path, is_engine_centric, use_sdk, should_build_tools_flag, has_monolithic, use_monolithic, expect_toolchain_build_called):
+    test_project_name = "TestProject"
+    test_project_path = tmp_path / "project"
+    test_engine_path = tmp_path / "engine"
+
+    test_o3de_base_path = test_project_path if not is_engine_centric else test_engine_path
+    test_relative_base_path = pathlib.PurePath('this/is/relative')
+    test_absolute_base_path  = tmp_path / "other"
+
+    assert test_absolute_base_path.is_absolute()
```

Comment:
Is this a necessary assert? isn't `tmp_path / "other"` ALWAYS an absolute path?

Reply:
I placed that there as a sanity check for me. I can remove it if need be, but it communicates to me that I'm certain I'm using an absolute path.